[scalar-field-filtering] Unexpected error when insert binary entities

**Describe the bug**
```
self = <test_delete.TestDeleteBase object at 0x7f7394ea6cc0>, connect = <milvus.client.stub.Milvus object at 0x7f7397fe1d30>, jac_collection = 'test_delete_PV89ilBs'

    @pytest.mark.level(2)
    def test_insert_same_ids_after_delete_jac(self, connect, jac_collection):
        '''
        method: add entities and delete
        expected: status DELETED
        '''
        insert_ids = [i for i in range(nb)]
        ids = connect.insert(jac_collection, binary_entities, insert_ids)
        connect.flush([jac_collection])
        delete_ids = [ids[0], ids[-1]]
        status = connect.delete_entity_by_id(jac_collection, delete_ids)
        assert status
>       new_ids = connect.insert(jac_collection, entity, [ids[0]])

test_delete.py:267: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/local/lib/python3.6/dist-packages/milvus/client/stub.py:35: in inner
    return func(self, *args, **kwargs)
/usr/local/lib/python3.6/dist-packages/milvus/client/stub.py:392: in insert
    return handler.insert(collection_name, entities, ids, partition_tag, params, timeout, **kwargs)
/usr/local/lib/python3.6/dist-packages/milvus/client/grpc_handler.py:50: in handler
    raise e
/usr/local/lib/python3.6/dist-packages/milvus/client/grpc_handler.py:42: in handler
    return func(self, *args, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <milvus.client.grpc_handler.GrpcHandler object at 0x7f7397fe17f0>, collection_name = 'test_delete_PV89ilBs'
entities = [{'field': 'int8', 'type': <DataType.INT8: 1>, 'values': [1]}, {'field': 'int64', 'type': <DataType.INT64: 4>, 'values...2709913, 0.1427215775477637, 0.1323921490526923, 0.09036368365407933, 0.03573359832352185, 0.07931000439912698, ...]]}]
ids = [0], partition_tag = None, params = None, timeout = None, kwargs = {}, insert_param = None
body = collection_name: "test_delete_PV89ilBs"
fields {
  field_name: "int8"
  attr_record {
    int32_value: 1
  }
}
fields ...77
      float_data: 0.12817463278770447
    }
  }
}
entity_id_array: 0
extra_params {
  key: "params"
  value: "{}"
}

rf = <_MultiThreadedRendezvous of RPC that terminated with:
	status = StatusCode.OK
	details = ""
>, response = status {
  error_code: UNEXPECTED_ERROR
  reason: "_Map_base::at"
}


    @error_handler([])
    def insert(self, collection_name, entities, ids=None, partition_tag=None, params=None, timeout=None, **kwargs):
        """
        Add vectors to collection
    
        :param ids: list of id
        :type  ids: list[int]
    
        :type  collection_name: str
        :param collection_name: collection name been inserted
    
        :type  records: list[list[float]]
    
                `example records: [[1.2345],[1.2345]]`
    
                `OR using Prepare.records`
    
        :param records: list of vectors been inserted
    
        :type partition_tag: str or None.
            If partition_tag is None, vectors will be inserted into collection rather than partitions.
    
        :param partition_tag: the tag string of collection
    
        :type
    
        :type  timeout: int
        :param timeout: time waiting for server response
    
        :returns:
            Status: indicate if vectors inserted successfully
            ids: list of id, after inserted every vector is given a id
        :rtype: (Status, list(int))
        """
        insert_param = kwargs.get('insert_param', None)
    
        if insert_param and not isinstance(insert_param, grpc_types.InsertParam):
            raise ParamError("The value of key 'insert_param' is invalid")
    
        body = insert_param if insert_param \
            else Prepare.insert_param(collection_name, entities, partition_tag, ids, params)
    
        # rf = self._stub.Insert.future(body, wait_for_ready=True, timeout=timeout)
        rf = self._stub.Insert.future(body, wait_for_ready=True, timeout=timeout)
        if kwargs.get("_async", False) is True:
            cb = kwargs.get("_callback", None)
            return InsertFuture(rf, cb)
    
        response = rf.result()
        if response.status.error_code == 0:
            return list(response.entity_id_array)
    
>       raise BaseException(response.status.error_code, response.status.reason)
E       milvus.client.exceptions.BaseException: <BaseException: (code=1, message=_Map_base::at)>

/usr/local/lib/python3.6/dist-packages/milvus/client/grpc_handler.py:431: BaseException
----------------------------------------------------------------------------------------------------------- Captured log setup -----------------------------------------------------------------------------------------------------------
[2020-07-07 17:20:31-INFO-root]: {'fields': [{'field': 'int8', 'type': <DataType.INT8: 1>}, {'field': 'int64', 'type': <DataType.INT64: 4>}, {'field': 'float', 'type': <DataType.FLOAT: 40>}, {'field': 'binary_vector', 'type': <DataType.BINARY_VECTOR: 101>, 'params': {'dimension': 128, 'metric_type': 'JACCARD'}}], 'segment_size': 10} (conftest.py:142)
---------------------------------------------------------------------------------------------------------- Captured stderr call ----------------------------------------------------------------------------------------------------------
Error: <BaseException: (code=1, message=_Map_base::at)>
----------------------------------------------------------------------------------------------------------- Captured log call ------------------------------------------------------------------------------------------------------------
[2020-07-07 17:20:31-ERROR-milvus.client.grpc_handler]: Error: <BaseException: (code=1, message=_Map_base::at)> (grpc_handler.py:44)

```

**Steps/Code to reproduce behavior**
Follow this [guide](http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) to craft a minimal bug report. This helps us reproduce the issue you're having and resolve the issue more quickly.

**Expected behavior**
A clear and concise description of what you expected to happen.

**Environment details**
- Hardware/Softward conditions (OS, CPU, GPU, Memory)
- Method of installation (Docker, or from source)
- Milvus version (v0.3.1, or v0.4.0)
- Milvus configuration (Settings you made in `server_config.yaml`)

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Additional context**
Add any other context about the problem here.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -27,6 +27,7 @@ Please mark all changes in change log and use the issue from GitHub
 -   \#2733 The max value of top-k should be limited
 -   \#2763 Unexpected error when insert binary entities
 -   \#2765 Server crashed when calling `get_entity_by_id`
+-   \#2783 Wrong result returned if searching with tags
 
 ## Feature
 -   \#2319 Redo metadata to support MVCC

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -25,6 +25,7 @@ Please mark all changes in change log and use the issue from GitHub
 -   \#2731 No entity returned with `get_entity_by_id`
 -   \#2732 Server destroyed after `delete by id`
 -   \#2733 The max value of top-k should be limited
+-   \#2763 Unexpected error when insert binary entities
 -   \#2765 Server crashed when calling `get_entity_by_id`
 
 ## Feature

--- a/core/src/db/DBImpl.cpp
+++ b/core/src/db/DBImpl.cpp
@@ -1340,6 +1340,10 @@ DBImpl::GetEntitiesByID(const std::string& collection_id, const milvus::engine::
         return status;
     }
 
+    if (field_names.empty()) {
+        return Status::OK();
+    }
+
     engine::meta::CollectionSchema collection_schema;
     engine::meta::hybrid::FieldsSchema fields_schema;
     collection_schema.collection_id_ = collection_id;

--- a/core/src/server/delivery/hybrid_request/HybridSearchRequest.cpp
+++ b/core/src/server/delivery/hybrid_request/HybridSearchRequest.cpp
@@ -101,12 +101,7 @@ HybridSearchRequest::OnExecute() {
             }
         }
 
-        //        if (field_names_.empty()) {
-        //            for (const auto& field : fields_schema.fields_schema_) {
-        //                field_names_.emplace_back(field.field_name_);
-        //            }
-        //        }
-
+        result_.row_num_ = 0;
         status = DBWrapper::DB()->HybridQuery(context_, collection_name_, partition_list_, general_query_, query_ptr_,
                                               field_names_, attr_type, result_);
 

--- a/core/src/server/delivery/hybrid_request/InsertEntityRequest.cpp
+++ b/core/src/server/delivery/hybrid_request/InsertEntityRequest.cpp
@@ -90,6 +90,21 @@ InsertEntityRequest::OnExecute() {
             }
         }
 
+        for (const auto& schema : fields_schema.fields_schema_) {
+            if (schema.field_type_ == (int32_t)engine::meta::hybrid::DataType::FLOAT_VECTOR &&
+                vector_datas_.at(schema.field_name_).float_data_.empty()) {
+                return Status{
+                    SERVER_INVALID_ROWRECORD_ARRAY,
+                    "The vector field is defined as float vector. Make sure you have entered float vector records"};
+            }
+            if (schema.field_type_ == (int32_t)engine::meta::hybrid::DataType::BINARY_VECTOR &&
+                vector_datas_.at(schema.field_name_).binary_data_.empty()) {
+                return Status{
+                    SERVER_INVALID_ROWRECORD_ARRAY,
+                    "The vector field is defined as binary vector. Make sure you have entered binary vector records"};
+            }
+        }
+
         std::unordered_map<std::string, engine::meta::hybrid::DataType> field_types;
         auto size = fields_schema.fields_schema_.size();
         for (const auto& field_name : field_names_) {

--- a/sdk/examples/simple/src/ClientTest.cpp
+++ b/sdk/examples/simple/src/ClientTest.cpp
@@ -35,6 +35,7 @@ constexpr int64_t SEARCH_TARGET = BATCH_ENTITY_COUNT / 2;  // change this value,
 constexpr int64_t ADD_ENTITY_LOOP = 1;
 constexpr milvus::IndexType INDEX_TYPE = milvus::IndexType::IVFFLAT;
 constexpr int32_t NLIST = 16384;
+constexpr char* PARTITION_TAG = "part";
 
 void
 PrintEntity(const std::string& tag, const milvus::VectorData& entity) {
@@ -209,7 +210,7 @@ ClientTest::SearchEntities(const std::string& collection_name, int64_t topk, int
 
     std::vector<std::string> partition_tags;
     milvus::TopKQueryResult topk_query_result;
-    auto status = conn_->Search(collection_name, partition_tags, dsl_json.dump(), vector_param, topk_query_result);
+    status = conn_->Search(collection_name, partition_tags, dsl_json.dump(), vector_param, topk_query_result);
 
     std::cout << "Search function call result: " << std::endl;
     milvus_sdk::Utils::PrintTopKQueryResult(topk_query_result);

